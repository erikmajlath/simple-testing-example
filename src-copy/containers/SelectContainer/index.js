import React from 'react'
import Select, { OPTIONS } from 'components/Select'

export default class SimpleSelect extends React.Component {
  state = {
    value: ['sf'],
  }

  onSelectChange = value => this.setState({ value })

  render() {
    return (
      <Select
        options={OPTIONS}
        value={this.state.value}
        onChange={this.onSelectChange}
      />
    )
  }
}
