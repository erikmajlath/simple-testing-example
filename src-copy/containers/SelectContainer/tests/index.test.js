import React from 'react'
import renderer from 'react-test-renderer'
import { createRenderer } from 'react-test-renderer/shallow'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import SelectContainer from '../index'

describe('SelectContainer', () => {
  const component = <SelectContainer />
  const PH = { value: 'ph', label: 'Prague' }
  let wrapper

  beforeEach(() => {
    wrapper = shallow(component)
  })

  it('should match snapshot', () => {
    expect(renderer.create(component).toJSON()).toMatchSnapshot()
  })

  it('should match snapshot shallowly', () => {
    const renderer = createRenderer()
    renderer.render(component)

    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })

  it('should update selected value', () => {
    wrapper.instance().onSelectChange('NICE')

    expect(wrapper.state('value')).toContain('NICE')
  })

  describe('renders according to state', () => {
    it('passes value to select', () => {
      wrapper.setState({ value: ['ph'] })

      expect(wrapper.find({ value: ['ph'] }).length).toBe(1)
    })

    it('matches snapshot with new state', () => {
      wrapper.setState({ value: ['ph'] })

      expect(toJson(wrapper)).toMatchSnapshot()
    })
  })
})
