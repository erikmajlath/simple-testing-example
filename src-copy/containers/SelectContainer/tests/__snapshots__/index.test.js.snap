// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectContainer renders according to state matches snapshot with new state 1`] = `
<Component
  onChange={[Function]}
  options={
    Array [
      Object {
        "label": "San Francisco",
        "value": "sf",
      },
      Object {
        "label": "Los Angeles",
        "value": "la",
      },
      Object {
        "label": "New York City",
        "value": "ny",
      },
    ]
  }
  value={
    Array [
      "ph",
    ]
  }
/>
`;

exports[`SelectContainer should match snapshot 1`] = `
<ul>
  <li
    onClick={[Function]}
  >
    All
  </li>
  <li
    onClick={[Function]}
  >
    San Francisco
     - SELECTED!
  </li>
  <li
    onClick={[Function]}
  >
    Los Angeles
  </li>
  <li
    onClick={[Function]}
  >
    New York City
  </li>
</ul>
`;

exports[`SelectContainer should match snapshot shallowly 1`] = `
<Unknown
  onChange={[Function]}
  options={
    Array [
      Object {
        "label": "San Francisco",
        "value": "sf",
      },
      Object {
        "label": "Los Angeles",
        "value": "la",
      },
      Object {
        "label": "New York City",
        "value": "ny",
      },
    ]
  }
  value={
    Array [
      "sf",
    ]
  }
/>
`;
